src/restic/node_unix_test.go

```diff
@@ -0,0 +1,119 @@
+// +build !windows
+
+package restic_test
+
+import (
+	"os"
+	"restic"
+	"syscall"
+	"testing"
+	"time"
+)
+
+func stat(t testing.TB, filename string) (fi os.FileInfo, ok bool) {
+	fi, err := os.Lstat(filename)
+	if err != nil && os.IsNotExist(err) {
+		return fi, false
+	}
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return fi, true
+}
+
+func checkFile(t testing.TB, stat *syscall.Stat_t, node *restic.Node) {
```

Comment:
The problem is rather that the things we're accessing and saving in our data structures are very different in different OS, and very low-level as well.